pkg/networkmanager/test-utils.js

```diff
@@ -0,0 +1,275 @@
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+var utils = require("./utils");
+var cockpit = require("cockpit");
+var QUnit = require("qunit-tests");
+var assert = QUnit;
+
+function assert_throws(func, checks) {
+    var i;
+
+    assert.expect(checks.length);
+
+    function doit(i) {
+        assert.throws(function () {
+            console.log(checks[i]);
+            func(checks[i]);
+        });
+    }
+
+    for (i = 0; i < checks.length; i++) {
+        doit(i);
+    }
+}
+
+QUnit.test("ip_prefix_from_text invalids", function() {
+    var checks = [
+        "",
+        "-1",
+        "foo",
+        "1foo",
+        "1.5"
+    ];
+
+    assert_throws(utils.ip_prefix_from_text, checks);
+});
+
+QUnit.test("ip_metric_from_text invalids", function() {
+    var checks = [
+        "-1",
+        "foo",
+        "1foo",
+        "1.5"
+    ];
+
+    assert_throws(utils.ip_metric_from_text, checks);
+});
+
+QUnit.test("ip4_to/from_text be", function() {
+    var i;
+
+    var checks = [
+        [ "0.0.0.0",           0x00000000 ],
+        [ "255.255.255.255",   0xFFFFFFFF ],
+        [ "1.2.3.4",           0x01020304 ]
+    ];
+
+    assert.expect(2*checks.length);
+
+    utils.set_byteorder("be");
+    for (i = 0; i < checks.length; i++) {
+        assert.strictEqual(utils.ip4_to_text(checks[i][1]), checks[i][0]);
+        assert.strictEqual(utils.ip4_from_text(checks[i][0]), checks[i][1]);
+    }
+});
+
+QUnit.test("ip4_to/from_text le", function() {
+    var i;
+
+    var checks = [
+        [ "0.0.0.0",           0x00000000 ],
+        [ "255.255.255.255",   0xFFFFFFFF ],
+        [ "1.2.3.4",           0x04030201 ]
+    ];
+
+    assert.expect(2*checks.length);
+
+    utils.set_byteorder("le");
+    for (i = 0; i < checks.length; i++) {
+        assert.strictEqual(utils.ip4_to_text(checks[i][1]), checks[i][0]);
+        assert.strictEqual(utils.ip4_from_text(checks[i][0]), checks[i][1]);
+    }
+});
+
+QUnit.test("ip4_from_text invalids", function() {
+    var checks = [
+        "",
+        "0",
+        "0.0",
+        "0.0.0",
+        "0.0.0.0.0",
+        "-1.2.3.4",
+        "foo",
+        "1.foo.3.4",
+        "1foo.2.3.4",
+        "1.2.3.400",
+        "1,2,3,4"
+    ];
+
+    assert_throws(utils.ip4_from_text, checks);
+});
+
+QUnit.test("ip4_to_text zero", function() {
+    utils.set_byteorder("be");
+    assert.strictEqual(utils.ip4_to_text(0, true), "");
+});
+
+QUnit.test("ip4_from_text empty", function() {
+    utils.set_byteorder("be");
+    assert.strictEqual(utils.ip4_from_text("", true), 0);
+});
+
+QUnit.test("ip4_prefix_from_text", function() {
```

Comment:
the actual code uses trim to ignore whitespace (e.g. [here](https://github.com/cockpit-project/cockpit/pull/5598/files#diff-27d9c791d3b36baa7fa523a84d2ca62eR200))
I recommend adding a few tests with extra whitespace in the individual parts so we know that the trimming works properly.